src/renderer/src/camera_controller.cpp

```diff
@@ -0,0 +1,29 @@
+#include "camera_controller.h"
+
+void SphericalCameraController::SetRadius(const float radius) {
+	radius_ = radius;
+	ReplaceCamera();
+}
+
+void SphericalCameraController::SetLongitude(const float phi) {
+	if (phi < 0 || pi2 < phi)
+		phi_ = phi - pi2 * (std::floor(phi / pi2));
+	else
+		phi_ = phi;
+	ReplaceCamera();
+}
+
+void SphericalCameraController::SetLatitude(const float theta) {
+	theta_ = theta;
+	ReplaceCamera();
+}
+
+void SphericalCameraController::ReplaceCamera() {
+	const float x_radius = std::cos(phi_) * std::cos(theta_) * radius_;
+	const float y_radius = std::sin(phi_) * std::cos(theta_) * radius_;
+	const float z_radius = std::sin(theta_) * radius_;
```

Comment:
mabe it's good idea to leave comment with some website with description of what is done

Reply:
For other branch with refactoring